instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerE2ETests.kt

```diff
@@ -250,4 +270,246 @@ class LoggerE2ETests {
+    fun logs_logger_add_date_null_attribute() {
+        val testMethodName = "logs_logger_add_date_null_attribute"
+        val nullDate: Date? = null
+        measure(testMethodName) {
+            logger.addAttribute(SPECIAL_DATE_ATTRIBUTE_NAME, nullDate)
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#fun removeAttribute(String)
+     */
+    @Test
+    fun logs_logger_remove_attribute() {
+        val testMethodName = "logs_logger_remove_attribute"
+        when (forge.anInt(min = 1, max = 9)) {
+            1 -> logger.addAttribute(
+                SPECIAL_ATTRIBUTE_NAME,
+                forge.aNullable { forge.anAlphabeticalString() }
+            )
+            2 -> logger.addAttribute(SPECIAL_ATTRIBUTE_NAME, forge.anInt())
+            3 -> logger.addAttribute(SPECIAL_ATTRIBUTE_NAME, forge.aFloat())
+            4 -> logger.addAttribute(SPECIAL_ATTRIBUTE_NAME, forge.aDouble())
+            5 -> logger.addAttribute(SPECIAL_ATTRIBUTE_NAME, forge.aBool())
+            6 -> logger.addAttribute(
+                SPECIAL_ATTRIBUTE_NAME,
+                forge.aNullable { CUSTOM_JSON_OBJECT_ATTRIBUTE }
+            )
+            7 -> logger.addAttribute(
+                SPECIAL_ATTRIBUTE_NAME,
+                forge.aNullable { CUSTOM_JSON_ARRAY_ATTRIBUTE }
+            )
+            8 -> logger.addAttribute(
+                SPECIAL_ATTRIBUTE_NAME,
+                forge.aNullable { CUSTOM_DATE_ATTRIBUTE }
+            )
+        }
+        measure(testMethodName) {
+            logger.removeAttribute(SPECIAL_ATTRIBUTE_NAME)
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#fun addTag(String, String)
+     */
+    @Test
+    fun logs_logger_add_tag() {
+        val testMethodName = "logs_logger_add_tag"
+        measure(testMethodName) {
+            logger.addTag(SPECIAL_TAG_NAME, CUSTOM_STRING_TAG)
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#fun addTag(String, String)
+     */
+    @Test
+    fun logs_logger_add_already_formatted_tag() {
+        val testMethodName = "logs_logger_add_already_formatted_tag"
+        measure(testMethodName) {
+            logger.addTag("$SPECIAL_TAG_NAME:$CUSTOM_STRING_TAG")
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#fun removeTag(String)
+     */
+    @Test
+    fun logs_logger_remove_tag() {
+        val testMethodName = "logs_logger_remove_tag"
+        if (forge.aBool()) {
+            logger.addTag(SPECIAL_TAG_NAME, CUSTOM_STRING_TAG)
+        } else {
+            logger.addTag("$SPECIAL_TAG_NAME:$CUSTOM_STRING_TAG")
+        }
+        measure(testMethodName) {
+            logger.removeTag("$SPECIAL_TAG_NAME:$CUSTOM_STRING_TAG")
+        }
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync()
```

Comment:
is this needed here?

Reply:
it should not be, but for some reason at some point this log event was not sent. I need to double check this.